src/libs/pr.ts

```diff
@@ -1,17 +1,50 @@
+// import * as core from '@actions/core';
 import { context, getOctokit } from '@actions/github';
 import { invariant } from './utils';
 
-export async function addPullRequestMessage(
+let octokit;
+
+export async function handlePullRequestMessage(
   body: string,
   githubToken: string,
 ): Promise<void> {
-  const { payload, repo } = context;
-  invariant(payload.pull_request, 'Missing pull request event data.');
+  invariant(context.payload.pull_request, 'Missing pull request event data.');
+  octokit = getOctokit(githubToken);
+
+  if (body && githubToken) {
+    if (context.payload.pull_request.number) {
+      await createCommentOnPullRequest(body);
+    }
+  }
+}
+
+async function createCommentOnPullRequest(body) {
+  const commentId = await findPreviousComments(`#### :tropical_drink: `);
 
-  const octokit = getOctokit(githubToken);
-  await octokit.issues.createComment({
-    ...repo,
-    issue_number: payload.pull_request.number,
-    body,
+  if (commentId) {
+    await octokit.issues.updateComment({
+      ...context.repo,
+      comment_id: commentId,
+      body,
+    });
+  } else {
+    await octokit.issues.createComment({
+      ...context.repo,
+      issue_number: context.payload.pull_request.number,
+      body,
+    });
+  }
+}
+
+async function getComments() {
+  return octokit.issues.listComments({
```

Comment:
Our [workflow](https://github.com/pulumi/actions/pull/148/checks?check_run_id=2223249099) is getting this error:

```shell
TypeError: octokit.issues.listComments is not a function
```

Would you mind checking that out?